build_release.sh

```diff
@@ -36,32 +36,47 @@ upx=`which upx`
 
 if [ ${#upx} -gt 0 ]; then
   echo "building and uploading ${projectname}-darwin-amd64-debug"
-  BUILDTIME=$(date -u '+%Y-%m-%d_%H:%M:%S') BUILDVERSION=$(git describe --tags) && env GOOS=darwin GOARCH=amd64 go build -ldflags "-X main.buildtime=$BUILDTIME_v${1} -X main.buildversion=${BUILDVERSION}" && date
+  BUILDTIME=$(date -u '+%Y-%m-%d_%H:%M:%S') BUILDVERSION=$(git describe --tags) && env GOOS=darwin GOARCH=amd64 go build -ldflags "-X main.buildtime=${BUILDTIME1} -X main.buildversion=${BUILDVERSION}" && date
   zip ${projectname}-darwin-amd64-debug.zip ${projectname}
   github-release upload     --user xorpaul     --repo ${projectname}     --tag v${1}     --name "${projectname}-darwin-amd64-debug.zip" --file ${projectname}-darwin-amd64-debug.zip
 fi
 
 echo "building and uploading ${projectname}-darwin-amd64"
-BUILDTIME=$(date -u '+%Y-%m-%d_%H:%M:%S') BUILDVERSION=$(git describe --tags) && env GOOS=darwin GOARCH=amd64 go build -ldflags "-s -w -X main.buildtime=$BUILDTIME_v${1} -X main.buildversion=${BUILDVERSION}" && date
+BUILDTIME=$(date -u '+%Y-%m-%d_%H:%M:%S') BUILDVERSION=$(git describe --tags) && env GOOS=darwin GOARCH=amd64 go build -ldflags "-s -w -X main.buildtime=${BUILDTIME} -X main.buildversion=${BUILDVERSION}" && date
 if [ ${#upx} -gt 0 ]; then
   $upx --brute ${projectname}
 fi
 zip ${projectname}-darwin-amd64.zip ${projectname}
 github-release upload     --user xorpaul     --repo ${projectname}     --tag v${1}     --name "${projectname}-darwin-amd64.zip" --file ${projectname}-darwin-amd64.zip
 
 
+if [ ${#upx} -gt 0 ]; then
```

Comment:
Why build the darwin binary again? They are being build just above this

Reply:
@xorpaul Apple Devices with M1 or M2 chip are using a arm CPU architecture instead of Intel/AMD (GOARCH=arm64 vs. GOARCH=amd64).